### ISSUE TITLE
csrfprotector.php uncaught error

### ISSUE THREAD

**Issue by kevinsmartpm** [nilsteampassnet/TeamPass#3234]
### Steps to reproduce
1. Fresh install of teampass
2. finish the initial setup for instalation
3. type localhost on url

### Expected behaviour
Tell us what should happen,
It should start the application

### Actual behaviour
Tell us what happens instead

![image](https://user-images.githubusercontent.com/108066583/175286541-a03de4f4-9a6f-4a40-ae4c-4f13191df441.png)

### Server configuration
**Operating system**: w10 pro

**Web server:** XAMPP

**Database:** MYSQL

**PHP version:** 8.1.6

**Teampass version:** [3.0.0.18]

**Teampass configuration file:**

**Updated from an older Teampass or fresh install:**
PLEASE attach to this issue the file `/includes/config/tp.config.php`.
 Fresh install 
<?php
global $SETTINGS;
$SETTINGS = array (
    'max_latest_items' => '10',
    'enable_favourites' => '1',
    'show_last_items' => '1',
    'enable_pf_feature' => '0',
    'log_connections' => '1',
    'log_accessed' => '1',
    'time_format' => 'H:i:s',
    'date_format' => 'd/m/Y',
    'duplicate_folder' => '0',
    'item_duplicate_in_same_folder' => '0',
    'duplicate_item' => '0',
    'number_of_used_pw' => '3',
    'manager_edit' => '1',
    'cpassman_dir' => 'C:/xampp/htdocs',
    'cpassman_url' => 'http://localhost',
    'favicon' => 'http://localhost/favicon.ico',
    'path_to_upload_folder' => 'C:/xampp/htdocs/upload',
    'path_to_files_folder' => 'C:/xampp/htdocs/files',
    'url_to_files_folder' => 'http://localhost/files',
    'activate_expiration' => '0',
    'pw_life_duration' => '0',
    'maintenance_mode' => '1',
    'enable_sts' => '0',
    'encryptClientServer' => '1',
    'cpassman_version' => '3.0.0.18',
    'ldap_mode' => '0',
    'ldap_type' => '0',
    'ldap_suffix' => '0',
    'ldap_domain_dn' => '0',
    'ldap_domain_controler' => '0',
    'ldap_user_attribute' => '0',
    'ldap_ssl' => '0',
    'ldap_tls' => '0',
    'ldap_elusers' => '0',
    'ldap_search_base' => '0',
    'ldap_port' => '389',
    'richtext' => '0',
    'allow_print' => '0',
    'roles_allowed_to_print' => '0',
    'show_description' => '1',
    'anyone_can_modify' => '0',
    'anyone_can_modify_bydefault' => '0',
    'nb_bad_authentication' => '0',
    'utf8_enabled' => '1',
    'restricted_to' => '0',
    'restricted_to_roles' => '0',
    'enable_send_email_on_user_login' => '0',
    'enable_user_can_create_folders' => '0',
    'insert_manual_entry_item_history' => '0',
    'enable_kb' => '0',
    'enable_email_notification_on_item_shown' => '0',
    'enable_email_notification_on_user_pw_change' => '0',
    'custom_logo' => '',
    'custom_login_text' => '',
    'default_language' => 'english',
    'send_stats' => '0',
    'send_statistics_items' => 'stat_country;stat_users;stat_items;stat_items_shared;stat_folders;stat_folders_shared;stat_admins;stat_managers;stat_ro;stat_mysqlversion;stat_phpversion;stat_teampassversion;stat_languages;stat_kb;stat_suggestion;stat_customfields;stat_api;stat_2fa;stat_agses;stat_duo;stat_ldap;stat_syslog;stat_stricthttps;stat_fav;stat_pf;',
    'send_stats_time' => '1653389714',
    'get_tp_info' => '1',
    'send_mail_on_user_login' => '0',
    'sending_emails' => '0',
    'nb_items_by_query' => 'auto',
    'enable_delete_after_consultation' => '0',
    'enable_personal_saltkey_cookie' => '0',
    'personal_saltkey_cookie_duration' => '31',
    'email_smtp_server' => '',
    'email_smtp_auth' => '',
    'email_auth_username' => '',
    'email_auth_pwd' => '',
    'email_port' => '',
    'email_security' => '',
    'email_server_url' => '',
    'email_from' => '',
    'email_from_name' => '',
    'pwd_maximum_length' => '40',
    'google_authentication' => '0',
    'delay_item_edition' => '0',
    'allow_import' => '0',
    'proxy_ip' => '',
    'proxy_port' => '',
    'upload_maxfilesize' => '10mb',
    'upload_docext' => 'doc,docx,dotx,xls,xlsx,xltx,rtf,csv,txt,pdf,ppt,pptx,pot,dotx,xltx',
    'upload_imagesext' => 'jpg,jpeg,gif,png',
    'upload_pkgext' => '7z,rar,tar,zip',
    'upload_otherext' => 'sql,xml',
    'upload_imageresize_options' => '1',
    'upload_imageresize_width' => '800',
    'upload_imageresize_height' => '600',
    'upload_imageresize_quality' => '90',
    'use_md5_password_as_salt' => '0',
    'ga_website_name' => 'TeamPass for ChangeMe',
    'api' => '0',
    'subfolder_rights_as_parent' => '0',
    'show_only_accessible_folders' => '0',
    'enable_suggestion' => '0',
    'otv_expiration_period' => '7',
    'default_session_expiration_time' => '60',
    'duo' => '0',
    'enable_server_password_change' => '0',
    'ldap_object_class' => '0',
    'bck_script_path' => 'C:/xampp/htdocs/backups',
    'bck_script_filename' => 'bck_teampass',
    'syslog_enable' => '0',
    'syslog_host' => 'localhost',
    'syslog_port' => '514',
    'manager_move_item' => '0',
    'create_item_without_password' => '0',
    'otv_is_enabled' => '0',
    'agses_authentication_enabled' => '0',
    'item_extra_fields' => '0',
    'saltkey_ante_2127' => 'none',
    'migration_to_2127' => 'done',
    'files_with_defuse' => 'done',
    'timezone' => 'UTC',
    'enable_attachment_encryption' => '1',
    'personal_saltkey_security_level' => '50',
    'ldap_new_user_is_administrated_by' => '0',
    'disable_show_forgot_pwd_link' => '0',
    'offline_key_level' => '0',
    'enable_http_request_login' => '0',
    'ldap_and_local_authentication' => '0',
    'secure_display_image' => '1',
    'upload_zero_byte_file' => '0',
    'upload_all_extensions_file' => '0',
    'bck_script_passkey' => 'hU3HWLkmNduWKHHF5GSsNyf6KArLZHfyxzRaxJk8',
    'admin_2fa_required' => '1',
    'password_overview_delay' => '4',
    'copy_to_clipboard_small_icons' => '1',
    'duo_akey' => '',
    'duo_ikey' => '',
    'duo_skey' => '',
    'duo_host' => '',
    'teampass_version' => '',
    'roles_allowed_to_print_select' => '',
    'clipboard_life_duration' => '30',
    'mfa_for_roles' => '',
    'tree_counters' => '0',
    'settings_offline_mode' => '0',
    'settings_tree_counters' => '0',
    'enable_massive_move_delete' => '0',
    'email_debug_level' => '0',
    'ga_reset_by_user' => '',
    'onthefly-backup-key' => '',
    'onthefly-restore-key' => '',
    'ldap_user_dn_attribute' => '',
    'ldap_dn_additional_user_dn' => '',
    'ldap_user_object_filter' => '',
    'ldap_bdn' => '',
    'ldap_hosts' => '',
    'ldap_password' => '',
    'ldap_username' => '',
    'api_token_duration' => '60',
    'last_folder_change' => '',
);

### Client configuration
**Browser:** chrome

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```
[Thu Jun 23 09:27:52.373938 2022] [ssl:warn] [pid 18840:tid 408] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 09:27:52.422827 2022] [ssl:warn] [pid 18840:tid 408] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 09:27:52.452529 2022] [mpm_winnt:notice] [pid 18840:tid 408] AH00455: Apache/2.4.53 (Win64) OpenSSL/1.1.1n PHP/8.1.6 configured -- resuming normal operations
[Thu Jun 23 09:27:52.452529 2022] [mpm_winnt:notice] [pid 18840:tid 408] AH00456: Apache Lounge VS16 Server built: Mar 16 2022 11:26:15
[Thu Jun 23 09:27:52.452529 2022] [core:notice] [pid 18840:tid 408] AH00094: Command line: 'C:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Thu Jun 23 09:27:52.464850 2022] [mpm_winnt:notice] [pid 18840:tid 408] AH00418: Parent: Created child process 6904
[Thu Jun 23 09:27:52.836547 2022] [ssl:warn] [pid 6904:tid 416] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 09:27:52.877251 2022] [ssl:warn] [pid 6904:tid 416] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 09:27:52.912131 2022] [mpm_winnt:notice] [pid 6904:tid 416] AH00354: Child: Starting 150 worker threads.
[Thu Jun 23 09:30:04.233644 2022] [ssl:warn] [pid 16708:tid 408] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 09:30:04.272899 2022] [ssl:warn] [pid 16708:tid 408] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
PHP Warning:  Module "openssl" is already loaded in Unknown on line 0
[Thu Jun 23 09:30:04.308554 2022] [mpm_winnt:notice] [pid 16708:tid 408] AH00455: Apache/2.4.53 (Win64) OpenSSL/1.1.1n PHP/8.1.6 configured -- resuming normal operations
[Thu Jun 23 09:30:04.308554 2022] [mpm_winnt:notice] [pid 16708:tid 408] AH00456: Apache Lounge VS16 Server built: Mar 16 2022 11:26:15
[Thu Jun 23 09:30:04.308554 2022] [core:notice] [pid 16708:tid 408] AH00094: Command line: 'c:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Thu Jun 23 09:30:04.323121 2022] [mpm_winnt:notice] [pid 16708:tid 408] AH00418: Parent: Created child process 16696
[Thu Jun 23 09:30:04.596151 2022] [ssl:warn] [pid 16696:tid 436] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 09:30:04.627690 2022] [ssl:warn] [pid 16696:tid 436] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
PHP Warning:  Module "openssl" is already loaded in Unknown on line 0
[Thu Jun 23 09:30:04.660973 2022] [mpm_winnt:notice] [pid 16696:tid 436] AH00354: Child: Starting 150 worker threads.
[Thu Jun 23 12:24:26.499922 2022] [ssl:warn] [pid 14868:tid 440] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 12:24:26.550977 2022] [core:warn] [pid 14868:tid 440] AH00098: pid file C:/xampp/apache/logs/httpd.pid overwritten -- Unclean shutdown of previous Apache run?
[Thu Jun 23 12:24:26.554045 2022] [ssl:warn] [pid 14868:tid 440] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
PHP Warning:  Module "gd" is already loaded in Unknown on line 0
PHP Warning:  Module "openssl" is already loaded in Unknown on line 0
[Thu Jun 23 12:24:26.603493 2022] [mpm_winnt:notice] [pid 14868:tid 440] AH00455: Apache/2.4.53 (Win64) OpenSSL/1.1.1n PHP/8.1.6 configured -- resuming normal operations
[Thu Jun 23 12:24:26.603493 2022] [mpm_winnt:notice] [pid 14868:tid 440] AH00456: Apache Lounge VS16 Server built: Mar 16 2022 11:26:15
[Thu Jun 23 12:24:26.603493 2022] [core:notice] [pid 14868:tid 440] AH00094: Command line: 'c:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Thu Jun 23 12:24:26.610294 2022] [mpm_winnt:notice] [pid 14868:tid 440] AH00418: Parent: Created child process 15004
[Thu Jun 23 12:24:26.903024 2022] [ssl:warn] [pid 15004:tid 412] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 12:24:26.946675 2022] [ssl:warn] [pid 15004:tid 412] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
PHP Warning:  Module "gd" is already loaded in Unknown on line 0
PHP Warning:  Module "openssl" is already loaded in Unknown on line 0
[Thu Jun 23 12:24:26.982491 2022] [mpm_winnt:notice] [pid 15004:tid 412] AH00354: Child: Starting 150 worker threads.
[Thu Jun 23 12:39:24.765936 2022] [php:warn] [pid 15004:tid 1884] [client ::1:60688] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165, referer: http://localhost/install/install.php
[Thu Jun 23 12:39:24.765936 2022] [php:error] [pid 15004:tid 1884] [client ::1:60688] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166, referer: http://localhost/install/install.php
[Thu Jun 23 12:39:45.413819 2022] [php:warn] [pid 15004:tid 1876] [client ::1:60699] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165
[Thu Jun 23 12:39:45.413819 2022] [php:error] [pid 15004:tid 1876] [client ::1:60699] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166
[Thu Jun 23 12:39:48.210853 2022] [php:warn] [pid 15004:tid 1876] [client ::1:60699] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165, referer: http://localhost/install/install.php
[Thu Jun 23 12:39:48.210853 2022] [php:error] [pid 15004:tid 1876] [client ::1:60699] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166, referer: http://localhost/install/install.php
[Thu Jun 23 12:39:51.580890 2022] [php:warn] [pid 15004:tid 1876] [client ::1:60699] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165, referer: http://localhost/install/install.php
[Thu Jun 23 12:39:51.580890 2022] [php:error] [pid 15004:tid 1876] [client ::1:60699] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166, referer: http://localhost/install/install.php
[Thu Jun 23 12:42:05.260458 2022] [php:warn] [pid 15004:tid 1884] [client ::1:60831] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165
[Thu Jun 23 12:42:05.261458 2022] [php:error] [pid 15004:tid 1884] [client ::1:60831] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166
[Thu Jun 23 12:47:26.444794 2022] [php:error] [pid 15004:tid 1876] [client ::1:60926] PHP Fatal error:  Uncaught mysqli_sql_exception: Unknown database 'tempass' in C:\\xampp\\htdocs\\install\\install.queries.php:277\nStack trace:\n#0 C:\\xampp\\htdocs\\install\\install.queries.php(277): mysqli_connect('localhost', 'root', 'Admin1234*', 'tempass', '3306')\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\install\\install.queries.php on line 277, referer: http://localhost/install/install.php
[Thu Jun 23 12:50:16.351079 2022] [php:error] [pid 15004:tid 1868] [client ::1:61005] PHP Fatal error:  Uncaught mysqli_sql_exception: Duplicate key name 'teampass_log_items_id_item_IDX' in C:\\xampp\\htdocs\\install\\install.queries.php:511\nStack trace:\n#0 C:\\xampp\\htdocs\\install\\install.queries.php(511): mysqli_query(Object(mysqli), 'CREATE INDEX te...')\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\install\\install.queries.php on line 511, referer: http://localhost/install/install.php
[Thu Jun 23 12:52:08.896505 2022] [php:error] [pid 15004:tid 1860] [client ::1:61077] PHP Fatal error:  Uncaught mysqli_sql_exception: Duplicate key name 'teampass_log_items_id_item_IDX' in C:\\xampp\\htdocs\\install\\install.queries.php:511\nStack trace:\n#0 C:\\xampp\\htdocs\\install\\install.queries.php(511): mysqli_query(Object(mysqli), 'CREATE INDEX te...')\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\install\\install.queries.php on line 511, referer: http://localhost/install/install.php
[Thu Jun 23 12:55:50.662595 2022] [php:warn] [pid 15004:tid 1860] [client ::1:61243] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165, referer: http://localhost/install/install.php
[Thu Jun 23 12:55:50.662595 2022] [php:error] [pid 15004:tid 1860] [client ::1:61243] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166, referer: http://localhost/install/install.php
[Thu Jun 23 12:57:42.752814 2022] [php:error] [pid 15004:tid 1868] [client ::1:61287] PHP Fatal error:  Can't use method return value in write context in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166, referer: http://localhost/install/install.php
[Thu Jun 23 12:58:17.981902 2022] [php:warn] [pid 15004:tid 1868] [client ::1:61297] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165
[Thu Jun 23 12:58:17.981902 2022] [php:error] [pid 15004:tid 1868] [client ::1:61297] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166
[Thu Jun 23 13:06:58.636096 2022] [ssl:warn] [pid 16816:tid 404] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 13:06:58.692453 2022] [core:warn] [pid 16816:tid 404] AH00098: pid file C:/xampp/apache/logs/httpd.pid overwritten -- Unclean shutdown of previous Apache run?
[Thu Jun 23 13:06:58.695455 2022] [ssl:warn] [pid 16816:tid 404] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
PHP Warning:  Module "gd" is already loaded in Unknown on line 0
PHP Warning:  Module "openssl" is already loaded in Unknown on line 0
[Thu Jun 23 13:06:58.749993 2022] [mpm_winnt:notice] [pid 16816:tid 404] AH00455: Apache/2.4.53 (Win64) OpenSSL/1.1.1n PHP/8.1.6 configured -- resuming normal operations
[Thu Jun 23 13:06:58.749993 2022] [mpm_winnt:notice] [pid 16816:tid 404] AH00456: Apache Lounge VS16 Server built: Mar 16 2022 11:26:15
[Thu Jun 23 13:06:58.749993 2022] [core:notice] [pid 16816:tid 404] AH00094: Command line: 'c:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Thu Jun 23 13:06:58.762863 2022] [mpm_winnt:notice] [pid 16816:tid 404] AH00418: Parent: Created child process 3000
[Thu Jun 23 13:06:59.079363 2022] [ssl:warn] [pid 3000:tid 416] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Thu Jun 23 13:06:59.123059 2022] [ssl:warn] [pid 3000:tid 416] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
PHP Warning:  Module "gd" is already loaded in Unknown on line 0
PHP Warning:  Module "openssl" is already loaded in Unknown on line 0
[Thu Jun 23 13:06:59.158667 2022] [mpm_winnt:notice] [pid 3000:tid 416] AH00354: Child: Starting 150 worker threads.
[Thu Jun 23 13:07:18.528006 2022] [php:warn] [pid 3000:tid 1888] [client ::1:62164] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165
[Thu Jun 23 13:07:18.528006 2022] [php:error] [pid 3000:tid 1888] [client ::1:62164] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166
[Thu Jun 23 13:12:37.972596 2022] [php:warn] [pid 3000:tid 1880] [client ::1:62335] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165
[Thu Jun 23 13:12:37.972596 2022] [php:error] [pid 3000:tid 1880] [client ::1:62335] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166
[Thu Jun 23 13:18:34.815126 2022] [php:warn] [pid 3000:tid 1888] [client ::1:62555] PHP Warning:  Trying to access array offset on value of type int in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 165
[Thu Jun 23 13:18:34.815126 2022] [php:error] [pid 3000:tid 1888] [client ::1:62555] PHP Fatal error:  Uncaught Error: Cannot use a scalar value as an array in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php:166\nStack trace:\n#0 C:\\xampp\\htdocs\\index.php(54): csrfProtector::init()\n#1 {main}\n  thrown in C:\\xampp\\htdocs\\includes\\libraries\\csrfp\\libs\\csrf\\csrfprotector.php on line 166

#### Log from the web-browser developer console (CTRL + SHIFT + i)
```
Insert the log here and especially the answer of the query that failed.
```

**Comment by nilsteampassnet** [comment]
Would it be possible to share `includes/libraries/csrfp/libs/csrfp.config.php`?

**Comment by useronkel** [comment]
I'm commenting the issue, same problem 
And to answer your question:
includes/libraries/csrfp/libs/csrfp.config.php is empty (0 bytes)
And to provide a more readable msg:
```
Notice: Trying to access array offset on value of type int in ./includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 165
Warning: Cannot use a scalar value as an array in ./includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 166
Notice: Trying to access array offset on value of type int in ./includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 169
Warning: Cannot use a scalar value as an array in ./includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 174
Notice: Trying to access array offset on value of type int in ./includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 177
Fatal error: Uncaught incompleteConfigurationException: OWASP CSRFProtector: Incomplete configuration file: missing failedAuthAction, jsUrl, tokenLength value(s) in ./includes/libraries/csrfp/libs/csrf/csrfprotector.php:188
Stack trace:
#0 ./index.php(54): csrfProtector::init()
#1 {main} thrown in ./includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 188
```
P.S.: Fresh install

**Comment by nilsteampassnet** [comment]
Oops missing sample file.
Did a mistake while updating the csrfp library during last commit.

Please update using latest commit.

**Comment by kevinsmartpm** [comment]
Thank you,

I Will try now!

De: Nils Laumaillé ***@***.***>
Enviado el: domingo, 26 de junio de 2022 18:54
Para: nilsteampassnet/TeamPass ***@***.***>
CC: Kevin Fernández ***@***.***>; Author ***@***.***>
Asunto: Re: [nilsteampassnet/TeamPass] csrfprotector.php uncaught error (Issue #3234)


Closed #3234<https://github.com/nilsteampassnet/TeamPass/issues/3234> as completed.
